### ISSUE TITLE
support unrar rar5.0 file?

### ISSUE THREAD

**Issue by sunpan3213** [lfuelling/lrkFM#68]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by lfuelling** [comment]
Hello, please don't just post the issue template as is. Please fill it out.

**Comment by lfuelling** [comment]
I did a little research. This project uses [commons-compress](http://commons.apache.org/proper/commons-compress/limitations.html) which doesn't even support rar < 5.

I could add rar < 5 support by using [junrar](https://github.com/junrar/junrar) but they also have no support for rar5. (see [#23](https://github.com/junrar/junrar/issues/23) and [#52](https://github.com/junrar/junrar/issues/52).

I could try using [Kaitai Struct](https://formats.kaitai.io/rar/index.html) (never heard of that project before) but as you can read in the junrar issues they also don't seem to support rar5.

I'll try getting it to work but I can't promise anything.

**Comment by sunpan3213** [comment]
all right



发自我的iPhone


------------------ Original ------------------

**Comment by sunpan3213** [comment]
ok



发自我的iPhone


------------------ Original ------------------

**Comment by lfuelling** [comment]
I know this doesn't fix the issue but I fixed support for RAR <v5.

**Comment by sunpan3213** [comment]
> 我做了一些研究。该项目使用[commons-compress](http://commons.apache.org/proper/commons-compress/limitations.html)，甚至不支持rar <5。
> 
> 我可以使用[junrar](https://github.com/junrar/junrar)添加rar <5支持，但它们也不支持rar5。（请参阅[＃23](https://github.com/junrar/junrar/issues/23)和[＃52](https://github.com/junrar/junrar/issues/52)）
> 
> 我可以尝试使用[Kaitai Struct](https://formats.kaitai.io/rar/index.html)（之前从未听说过该项目），但正如您在junrar问题中所读到的那样，它们似乎也不支持rar5。
> 
> 我会尽力使它正常工作，但是我什么都不能保证。

can you add some cpp lib to resolve this problem?

**Comment by sunpan3213** [comment]
> 我做了一些研究。该项目使用[commons-compress](http://commons.apache.org/proper/commons-compress/limitations.html)，甚至不支持rar <5。
> 
> 我可以使用[junrar](https://github.com/junrar/junrar)添加rar <5支持，但它们也不支持rar5。（请参阅[＃23](https://github.com/junrar/junrar/issues/23)和[＃52](https://github.com/junrar/junrar/issues/52)）
> 
> 我可以尝试使用[Kaitai Struct](https://formats.kaitai.io/rar/index.html)（之前从未听说过该项目），但正如您在junrar问题中所读到的那样，它们似乎也不支持rar5。
> 
> 我会尽力使它正常工作，但是我什么都不能保证。

i see,java can't support rar5.0

**Comment by sunpan3213** [comment]
> I did a little research. This project uses [commons-compress](http://commons.apache.org/proper/commons-compress/limitations.html) which doesn't even support rar < 5.
> 
> I could add rar < 5 support by using [junrar](https://github.com/junrar/junrar) but they also have no support for rar5. (see [#23](https://github.com/junrar/junrar/issues/23) and [#52](https://github.com/junrar/junrar/issues/52))
> 
> I could try using [Kaitai Struct](https://formats.kaitai.io/rar/index.html) (never heard of that project before) but as you can read in the junrar issues they also don't seem to support rar5.
> 
> I'll try getting it to work but I can't promise anything.

can you add some cpp lib to resolve this problem?

**Comment by lfuelling** [comment]
Yes, that's what I'm currently trying in #70 but getting it to build seems more complicated than I thought.
